api/src/routes/challenge.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  Type,
+  type FastifyPluginCallbackTypebox
+} from '@fastify/type-provider-typebox';
+import jwt from 'jsonwebtoken';
+import { uniqBy } from 'lodash';
+import { jwtSecret } from '../../../config/secrets';
+import { fixPartiallyCompletedChallengeItem } from '../utils';
+import { getChallenges } from '../utils/get-challenges';
+import { updateUserChallengeData } from '../utils/common-challenge-functions';
+interface JwtPayload {
+  userToken: string;
+}
+
+export const challengeRoutes: FastifyPluginCallbackTypebox = (
+  fastify,
+  _options,
+  done
+) => {
+  const challenges = getChallenges();
+  fastify.post(
+    '/challenge/coderoad-challenge-completed',
+    {
+      schema: {
+        body: Type.Object({
+          tutorialId: Type.String()
+        }),
+        response: {
+          200: Type.Object({ msg: Type.String() }),
+          400: Type.Object({
+            msg: Type.String()
+          }),
+          500: Type.Object({
+            message: Type.Literal('Something went wrong.'),
+            type: Type.Literal('danger')
+          })
+        }
+      },
+      attachValidation: true
+    },
+    async (req, reply) => {
+      let userToken;
+
+      const { 'coderoad-user-token': encodedUserToken } = req.headers;
+      const { tutorialId } = req.body;
+
+      if (!tutorialId) {
+        void reply.code(400);
+        return { msg: `'tutorialId' not found in request body` };
+      }
+
+      if (!encodedUserToken) {
+        void reply.code(400);
+        return { msg: `'coderoad-user-token' not found in request headers` };
+      }
+
+      try {
+        if (typeof encodedUserToken === 'string' && jwtSecret) {
+          const payload = jwt.verify(encodedUserToken, jwtSecret) as JwtPayload;
+          userToken = payload.userToken;
+        }
+      } catch {
+        void reply.code(400);
+        return { msg: `invalid user token` };
+      }
+
+      const tutorialRepo = tutorialId?.split(':')[0];
+      const tutorialOrg = tutorialRepo?.split('/')?.[0];
+
+      if (tutorialOrg !== 'freeCodeCamp') {
+        void reply.code(400);
+        return { msg: `Tutorial not hosted on freeCodeCamp GitHub account` };
+      }
```

Comment:
@moT01 I'm pretty sure you said something like this to me the other day, but just to confirm: can we get rid of the `if (deploymentEnv !== 'staging' && environment !== 'development') {` conditional in api-server? It looks like this is a leftover from early testing, but I wanted to confirm.

Reply:
Yes

Reply:
Thanks, I'll create a quick PR for that.